File: frontend-profitfill/src/models/job.ts
export type Job = {
    id: number;
    customerName: string;
    jobType: string;
    status: string;
    appointmentDate: string;
    technician: string;
};

export interface JobFormState {
    id: string;
    customerName: string;
    jobType: string;
    status: string;
    appointmentDate: string;
    technician: string;
}

export interface JobFormErrors {
    id?: string;
    customerName?: string;
    jobType?: string;
    status?: string;
    appointmentDate?: string;
    technician?: string;
}

File: backend-profitFill/Model/Job.cs
using System.Globalization;
using backend_profitFill.DTO;

namespace backend_profitFill.Model;

public class Job
{
 
 public static Job CreateInstance(JobDTO jobDto)
 {
  return new Job
  {
   Id = jobDto.Id,
   JobType = jobDto.JobType,
   CustomerName = jobDto.CustomerName,
   Status =  jobDto.Status,
   AppointmentDate = DateTime.ParseExact(jobDto.AppointmentDate, "yyyy-MM-ddTHH:mm:ssZ",  CultureInfo.InvariantCulture,
    DateTimeStyles.AdjustToUniversal),
   Technician = jobDto.Technician
  };
 }
 
 public void UpdateInstance(Job job)
 {
  if (job == null)
   throw new ArgumentNullException(nameof(job), "Source cannot be null.");
  
  this.JobType = job.JobType;
  this.CustomerName = job.CustomerName;
  this.Status = job.Status;
  this.AppointmentDate = job.AppointmentDate;
  this.Technician = job.Technician;
 }
 
 public int Id { get; set; }
 public string CustomerName { get; set; }
 public string JobType { get; set; }
 public string Status { get; set; }
 public DateTime AppointmentDate { get; set; }
 public string Technician { get; set; }
}